### DISCUSSION TITLE
Asterisk 20.6 on Ubuntu 24.04: Fail2Ban not working

### DISCUSSION THREAD

**Discussion by datonet17** [fail2ban/fail2ban#3836] [General]
Good day all!

I am not so familiar with Fail2Ban but is seems like changes on the new version, currently v1.0.2 installed, is giving me headache.
It looks like it is not using the .log files of asterisk anymore, but rather journal entries.

    Status for the jail: asterisk-iptables
    |- Filter
    | |- Currently failed: 0
    | |- Total failed: 0
    | - Journal matches: _SYSTEMD_UNIT=asterisk.service - Actions
    |- Currently banned: 0
    |- Total banned: 0
    - Banned IP list: ````Preformatted text

However it is not catching anything…
Where do I start?

Thank you!

**Comment by sebres** [comment]
> Where do I start?

See the answer for first Q in this [wiki :: FAQ](https://github.com/fail2ban/fail2ban/wiki/How-fail2ban-works).

And to help you here, we'd need an excerpt from journal or log-file with the messages that must be banned (better few lines).

**Reply by datonet17** [reply]
Thanks for reaching out!

The ubuntu journal shows no asterisk errors, and it seems like Fail2Ban is now looking only there to find failed attempts.
(Due to the fact that it shows "Journal matches: _SYSTEMD_UNIT=asterisk.service - Actions"

Previous Fail2Ban versions showed the log path there, instead of "Journal matches".

The Asterisk log located at /var/log/asterisk/messages.log shows the failed login attempts as expected.

Config at /etc/fail2ban/jail.conf includes the log file path:
```
[asterisk]

port     = 5060,5061
action_  = %(default/action_)s[name=%(__name__)s-tcp, protocol="tcp"]
           %(default/action_)s[name=%(__name__)s-udp, protocol="udp"]
logpath  = /var/log/asterisk/messages
maxretry = 10

[asterisk-iptables]

enabled = true
filter = asterisk
action = iptables-allports[name=ASTERISK, protocol=all]
logpath = /var/log/asterisk/messages
maxretry = 5
bantime = 259200
```

But fail2ban-client status asterisk-iptables show no mention of the log file and seems to ignore "logpath":
```
Status for the jail: asterisk-iptables
|- Filter
|  |- Currently failed: 0
|  |- Total failed:     0
|  `- Journal matches:  _SYSTEMD_UNIT=asterisk.service
```

I have not changed anything at /etc/fail2ban/filter.d/asterisk.conf

It seems like I either have to enable logging of asterisk to journal, or like the old method get Fail2Ban to read the log file again...
What should I do?

Thank you!

**Reply by sebres** [reply]
> Previous Fail2Ban versions showed the log path there, instead of "Journal matches".

Probably the maintainer of your distro decided to change default `backend` to `systemd`.

> What should I do?

Add `backend = auto` to the jail(s), as a newline for instance before `logpath`. That's it.
To check whether the filter basically works, you can use (see whether it'd show matches):
```bash
# statistic of the test (matches, misses, etc):
fail2ban-regex /var/log/asterisk/messages asterisk
# messages matched as failures:
fail2ban-regex -o msg /var/log/asterisk/messages asterisk
```

**Reply by datonet17** [reply]
Hi sebres,

Thank you so much for guidance!

Adding ```backend = auto``` resulted in ```Failed during configuration: Have not found any log file for asterisk-iptables jail``` during service restart.

So I had to add .log at the end of the logpath's:

```
[asterisk]

...
logpath  = /var/log/asterisk/messages.log
...

[asterisk-iptables]

...
backend = auto
logpath = /var/log/asterisk/messages.log
...
```

Not sure if it the best method, however I can already see it is working:
```
Status for the jail: asterisk-iptables
|- Filter
|  |- Currently failed: 1
|  |- Total failed:     7
|  `- File list:        /var/log/asterisk/messages.log
`- Actions
   |- Currently banned: 1
   |- Total banned:     1
   `- Banned IP list:   208.77.45.4
```

So much to learn! Thank you for taking the time.

**Reply by sebres** [reply]
> Not sure if it the best method

Why? Parameter `logpath` must point to the full path of log-file... basically it is a list (multiline) of glob, but if logfile ends with `.log`, it must surely ends with `.log` (or `*`). So it is not only the best but solely correct way.

**Reply by datonet17** [reply]
You are correct, upon further investigation it seems like asterisk changed their log name from ```messages``` tot ```messages.log```. (It might have been me in the process of trying to figure it out) However your logic makes sense and all is sorted!

Thank you!
